### ISSUE TITLE
Edge browser crash + shake label not working

### ISSUE THREAD

**Issue by kenpb** [trimox/angular-mdc-web#1740]
### What Angular MDC version are you using?
0.43.3

### What browser(s) is this bug affecting?
All major browsers, up to date versions

### What OS are you using?
Linux Mint 19, Windows 10 17763.253

### What are the steps to reproduce the bug?
In an outlined text field (non required) of a reactive form with any pattern validation; enter an invalid value, then, focus something else, the error styling should be applied; focus the control again, blur it, the styling should disappear as if the text field were valid.

When running on MS Edge (randomly; sometimes at the first blur, sometimes at the second) type and if the pattern has failed the browser will crash and refresh the page, this with required text fields.

### What is the expected behavior?
The error style should be always visible while the error is present.

### What is the actual behavior?
If you focus and blur an outlined text field with an error caused by a pattern validation, the style will be lost.

### Any other information you believe would be useful?
The style works as expected if the text field is required.

May be related to  #1606 and the crash only happening under `--prod` build.

~~I'll attach a link to an example soon.~~
Demo: https://kenpb.github.io/ng-edge-crash/
Source: https://github.com/kenpb/ng-edge-crash/tree/master/mdc-demo/app

The Edge crash happens after what it seems after two errors in a row.

**Comment by trimox** [comment]
@kenpb Once again thank you for reporting, and providing a demo! 💎 

Looks like ValidityState `patternMismatch` value is not respected somewhere. Let ya know

**Comment by trimox** [comment]
Got this resolved, and you also stumbled on a performance issue. PR incoming

**Comment by kenpb** [comment]
Awesome! Thank you again for the great support!

**Comment by kenpb** [comment]
Hey @trimox!

Did you come across the MS Edge crash I mentioned?, just tried `0.43.4` and now it happens with all inputs not only non-required, I've tried to debug it but the crash kills the console too, sadly I don't have much info on it.

**Comment by kenpb** [comment]
@trimox yeah, the dev build seems to be ok, here's a gif of what happens, as soon as it get out of focus it seems like it loops until the page crashes

![crash](https://user-images.githubusercontent.com/2224388/51570083-c196ec80-1e63-11e9-8054-ad261cf590db.gif)

**Comment by trimox** [comment]
@kenpb Getting closer. Removed `[helperText]="dateHelper3"`, and it's fine. Hrmm

**Comment by trimox** [comment]
@kenpb Strike that, soon as I focused again, and kept typing it crashed.

**Comment by kenpb** [comment]
@trimox Yeah it's kinda random, it used to took about two tries it took me a while to notice it.

**Comment by trimox** [comment]
Diving into the foundation call, specifically this line

https://github.com/material-components/material-components-web/blob/master/packages/mdc-textfield/foundation.js#L277

**Comment by trimox** [comment]
So, I'm pretty darn sure it's a side-effect of the Angular CLI `clean-css` process. I noticed in production builds that the shake label doesn't shake. FireFox and Chrome handle the missing css gracefully, unlike their counterpart Edge. Now to figure out a workaround..

**Comment by trimox** [comment]
Yep, confirmed. 

I changed the `cleancss-webpack-plugin.js` CLI config file for `clean-css` level from 2 to 0; worked perfectly.

```
            const cleancss = new CleanCSS({
                compatibility: 'ie9',
                level: {
                    0: {
                        skipProperties: ['transition'] // Fixes #12408
                    }
                },
                inline: false,
                returnPromise: true,
                sourceMap: this._options.sourceMap,
            });
```

**Comment by kenpb** [comment]
Hello @trimox, thanks for the follow ups mate, you're a wizard!

And I see, that makes sense, but doesn't that mean that ejecting will be the only workaround?

**Comment by trimox** [comment]
@kenpb Hah, no worries. Slept on it, and researched what we needed. PR incoming 

Thanks once again for reporting this! You saved a bunch of people tireless hours of debugging 💎
